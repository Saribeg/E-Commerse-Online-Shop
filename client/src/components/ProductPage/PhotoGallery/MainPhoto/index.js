import React, { Component, Fragment } from "react";

export default class MainPhoto extends Component {
  render() {
    return (
        <div className="photo-main"><img src={this.props.activePhotoSrc} alt="img" /></div>
    );
  }
}
