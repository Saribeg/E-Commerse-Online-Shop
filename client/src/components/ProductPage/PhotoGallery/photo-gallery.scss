@import "../../../scss/resources/resources";

.all-photos {
  display: flex;
  flex-direction: column;
  margin-right: 30px;

  @media (max-width:680px) {
    flex-direction: row;
    justify-content: space-between;
    overflow-x: auto;
    margin-top: 24px;
    margin-right: 0;
    order: 2;
  }

  &-item {
    width: 120px;
    height: 120px;
    margin-bottom: 20px;
    cursor: pointer;

    img {
      @include image;
    }

    @media (max-width:680px) {
      margin-right: 12px;
    }

    &.active {
      border: 2px solid $red;
    }
  }
}

.photo-main {
  max-width: 450px;
  max-height: 680px;
  //margin: 0 30px;

  @media (max-width:680px) {
    width: 100%;
    //max-width: 450px;
    //height: 100vh;
    max-height: 680px;
    //opacity: 0.2;
    margin: 0 auto;
    order: 1;
  }

  .all-photos-item {
    @include image;
    margin: 0;

    img {
      object-fit: cover;
    }
  }
}