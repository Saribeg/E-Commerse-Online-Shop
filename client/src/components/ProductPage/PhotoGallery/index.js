import React, {Component} from "react";
import {connect} from "react-redux";
import {SET_COLOR, SET_IMG} from "../../../actions/addToCart";

import "./photo-gallery.scss";


class PhotoGallery extends Component {
    state = {
        index: 0
    };

    // componentDidMount() {
    //
    //     console.log('componentDidMount GALLERY');
    //     console.log(this.props.activeColor)
    //
    //     this.props.setColorAddCart(this.props.activeColor);
    //
    // }

    componentDidUpdate(prevProps, prevState) {

        let urlPhoto = '';
        this.props.productFeatures.forEach((elem => {

            if (this.props.activeColor === elem.colorName) {
                urlPhoto = elem.imageUrls[0];
            }

        }));

        this.props.setColorAddCart(this.props.activeColor);
        this.props.setUrlAddCart(urlPhoto);

    }

    render() {
        let productFeatures = this.props.productFeatures;
        let photoGallery = productFeatures.map(elem => {
            let active = this.props.activeColor === elem.colorName;
            if (active) {
                let array = elem.imageUrls;
                return elem.imageUrls.map(elem => {
                    return (
                        <div className={`all-photos-item`} key={elem}>
                            <img

                                src={elem}
                                alt={this.props.activeColor}
                                onClick={() => {
                                    this.setState({
                                        index: array.indexOf(elem)
                                    });
                                }}
                            /></div>
                    );
                });
            }
        });

        let photoGalleryFiltered = photoGallery.filter(elem => {
            return elem !== undefined;
        })[0];

        let mainPhoto = (photoGalleryFiltered) ? photoGalleryFiltered[this.state.index] : null;

        return (
            <>
                <div className="all-photos">{photoGallery}</div>
                <div className="photo-main">
                    {mainPhoto}
                </div>
            </>
        );
    }
}

const mapStateToProps = state => {
    return {};
};

const mapDispatchToProps = dispatch => {
    return {
        // setInitialAddCart: (data) => {dispatch({type: SET_INITIAL_STATE, payload: {data: data}})}
        setColorAddCart: (color) => {
            dispatch({type: SET_COLOR, payload: {color: color}})
        },

        setUrlAddCart: (url) => {
            dispatch({type: SET_IMG, payload: {url: url}})
        }
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(PhotoGallery);
