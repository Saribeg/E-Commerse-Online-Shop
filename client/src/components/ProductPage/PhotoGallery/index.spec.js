import React from 'react';
import PhotoGallery from './index';

const productFeatures = [{
	"color": "#000000",
	"colorName": "black",
	"imageUrls": [
		"/img/products/women/clothing/hoodies/002/000000/021.jpg",
		"/img/products/women/clothing/hoodies/002/000000/022.jpg",
		"/img/products/women/clothing/hoodies/002/000000/023.jpg",
		"/img/products/women/clothing/hoodies/002/000000/024.jpg"
	],
	"sizes": [{
			"size": "xs",
			"quantity": 0
		},
		{
			"size": "s",
			"quantity": 0
		},
		{
			"size": "m",
			"quantity": 18
		},

		{
			"size": "l",
			"quantity": 23
		}
	]
}];

describe('PhotoGallery component', () => {
    it('PhotoGallery component render', () => {
        const wrapper  = shallow(<PhotoGallery productFeatures={productFeatures} />)
        expect(wrapper).toMatchSnapshot()
    })
})