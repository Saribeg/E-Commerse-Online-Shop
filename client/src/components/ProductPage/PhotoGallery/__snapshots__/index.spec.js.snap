// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGallery component PhotoGallery component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhotoGallery
    productFeatures={
      Array [
        Object {
          "color": "#000000",
          "colorName": "black",
          "imageUrls": Array [
            "/img/products/women/clothing/hoodies/002/000000/021.jpg",
            "/img/products/women/clothing/hoodies/002/000000/022.jpg",
            "/img/products/women/clothing/hoodies/002/000000/023.jpg",
            "/img/products/women/clothing/hoodies/002/000000/024.jpg",
          ],
          "sizes": Array [
            Object {
              "quantity": 0,
              "size": "xs",
            },
            Object {
              "quantity": 0,
              "size": "s",
            },
            Object {
              "quantity": 18,
              "size": "m",
            },
            Object {
              "quantity": 23,
              "size": "l",
            },
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="all-photos"
        />,
        <div
          className="photo-main"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
          ],
          "className": "all-photos",
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "photo-main",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="all-photos"
          />,
          <div
            className="photo-main"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
            ],
            "className": "all-photos",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "photo-main",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
